lib/service/modules/entity.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (c) 2016 Kinvey Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+const moment = require('moment');
+const BSON = require('bson').BSONPure;
+
+module.exports = (environmentId, useBSONObjectId) => {
+  if (!useBSONObjectId) useBSONObjectId = false;
+
+  const _generateKinveyEntity = (entityOrId) => {
+    let _entity = {};
+
+    const _createEntityId = () => {
+      if (useBSONObjectId) return new BSON.ObjectID();
+      return new BSON.ObjectID().toString();
+    };
+
+    const currentTimeString = moment().utc().toISOString();
+
+    if (typeof entityOrId === 'string' || typeof entityOrId === 'undefined') {
+      _entity = {
+        _acl: {
+          creator: environmentId
+        },
+        _kmd: {
+          lmt: currentTimeString,
+          ect: currentTimeString
+        }
+      };
+
+      if (entityOrId) {
+        _entity._id = entityOrId;
+      } else {
+        _entity._id = _createEntityId();
+      }
+    } else {
+      if (typeof entityOrId === 'object') {
```

Comment:
Can `entityOrId` be `null`? if so, you should handle that (since `typeof null === 'object'`)